sharding/indexHashedNodesCoordinator.go

```diff
@@ -454,14 +486,27 @@ func (ihgs *indexHashedNodesCoordinator) EpochStartAction(hdr data.HeaderHandler
 	ihgs.mutNodesConfig.Unlock()
 }
 
+// GetSavedStateKey returns the key for the last nodes coordinator saved state
+func (ihgs *indexHashedNodesCoordinator) GetSavedStateKey() []byte {
+	ihgs.mutSavedStateKey.RLock()
+	key := ihgs.savedStateKey
+	ihgs.mutSavedStateKey.RUnlock()
+
+	return key
+}
+
 func (ihgs *indexHashedNodesCoordinator) expandEligibleList(validators []Validator, mut *sync.RWMutex) []Validator {
 	//TODO implement an expand eligible list variant
 	return validators
 }
 
 func (ihgs *indexHashedNodesCoordinator) computeShardForPublicKey(nodesConfig *epochNodesConfig) uint32 {
 	pubKey := ihgs.selfPubKey
```

Comment:
I just discussed with Iulian and probably the best option, to avoid any edge cases, would be to have a configuration parameter for the shard to observe when no longer acting as validator. Will be added in another pull request.